pkg/yurthub/util/util.go

```diff
@@ -433,3 +434,42 @@ func CreateKubeConfigFile(kubeClientConfig *rest.Config, kubeconfigPath string)
 	// Marshal to disk
 	return clientcmd.WriteToFile(kubeconfigData, kubeconfigPath)
 }
+
+// gzipReaderCloser will un-gzip the data if response header
+// contains Content-Encoding=gzip header.
+type gzipReaderCloser struct {
+	body io.ReadCloser
+	zr   *gzip.Reader
+	zerr error
+}
+
+func (grc *gzipReaderCloser) Read(b []byte) (n int, err error) {
+	if grc.zerr != nil {
+		return 0, grc.zerr
+	}
+
+	if grc.zr == nil {
+		grc.zr, err = gzip.NewReader(grc.body)
+		if err != nil {
+			grc.zerr = err
+			return 0, err
+		}
+	}
+
+	return grc.zr.Read(b)
+}
+
+func (grc *gzipReaderCloser) Close() error {
+	return grc.body.Close()
```

Comment:
I think we'd better call `grc.zr.Close` before calling `grc.body.Close`

Reply:
`grc.zr.Close` is used to verify the reader is fully consumed or not, we need to close the underlying `io.Reader` here, so i think we only need to call `grc.body.Close()` here.

BTW: if we add `grc.zr.Close`, i don't how to handle the return error of `grc.zr.Close` now.

Reply:
OK, i'm not familiar with the gzip package. lgtm for this case.

Just in terms of how to handle the error, we can do it as follows
```go
func (grc *gzipReaderCloser) Close() error {
	if err := grc.zr.Close(); err != nil {
		klog.Errorf("info here")
	}
	return grc.body.Close()
```

Reply:
The `Close` method is used to verify the GZIP checksum:
```
// Close closes the Reader. It does not close the underlying io.Reader.
// In order for the GZIP checksum to be verified, the reader must be
// fully consumed until the io.EOF.
func (z *Reader) Close() error { return z.decompressor.Close() }
```
But I believe we don't need to call `grc.zr.Close` anymore, because this method now actually do nothing:
https://github.com/golang/go/blob/3e387528e54971d6009fe8833dcab6fc08737e04/src/compress/flate/inflate.go#L354-L359
```
func (f *decompressor) Close() error {
    if f.err == io.EOF {
        return nil
    }
    return f.err
}
```
 If the checksum is invalid, the `Read` method will set `f.err` and return it. I think we don't need a redundant check in `gzipReaderCloser.Close`.  

So I think `grc.body.Close()` is ok.

Reference: 
https://stackoverflow.com/questions/68074107/what-is-the-point-of-gzip-reader-close

Reply:
Thanks, it's reasonable.